### DISCUSSION TITLE
How to deal with exponential matrices in firedrake?

### DISCUSSION THREAD

**Discussion by zhyxue0** [firedrakeproject/firedrake#3230] [Firedrake support]
Hello, everyone!
I try to use ETD(exponential time differencing) time discreteness in firedrake. But I have some problems.
ETD time discretization involves processing exponential matrices `exp^(-tL)` ( `L` is the identity matrix, or discretized by the Laplacian operator).
How to deal with exponential matrices in firedrake? Are there any examples that I can refer to?
Thanks a lot!

**Comment by danshapero** [comment]
I don't know that Firedrake can do this directly, but you can get out the PETSc matrices and then call out to SLEPc for evaluating matrix exponentials. They have [some examples](https://slepc.upv.es/documentation/current/src/mfn/tutorials/index.html) but they're in C, so you'd need to translate them to use the Python API.
